### ISSUE TITLE
[Non-Urgent] Using Poetry to manage the project, simplify impl. of install, dep. and unit test config

### ISSUE THREAD

**Issue by DouglasKrouth** [Police-Data-Accessibility-Project/data-sources-app#73]
This would be an easy conversion from the current format outlined in the README.md that uses requirements.txt.

Would allow us simpler setup and configuration of the data-sources-app. Happy to do a walkthrough/meeting to chat about this. 

Primary benefits would be the introduction of a python .lock file, pyproject.toml for simplifies implementing a pytest suite, and ease of setup for incoming volunteers/devs so long as they're willing to install the Poetry dependency. 

Would convert the current install process into simply cloning the repo and running ```poetry install```

Poetry : https://python-poetry.org/docs/

Reached out to Josh about getting access to branching on the repo, once that's available I can upload what it would look like and y'all can review to see if you'd like to go this route.

**Comment by josh-chamberlain** [comment]
This is generally over my head... @mbodeantor ?

**Comment by DouglasKrouth** [comment]
@mbodeantor Sample of what it'd look like to migrate to Poetry. Just adds a pyproject.toml and a lock file. Lock file would enable much more consistent builds in prod (similar to cargo.lock or npm-style) and it makes uses pytest a bit easier.

One important note: we could just the pyproject.toml without Poetry; however, we wouldn't have the dependency resolution layer built on top of ```pip``` or the use of a .lock file.

[branch: poetry_impl](https://github.com/Police-Data-Accessibility-Project/data-sources-app/tree/poetry_impl)

**Comment by josh-chamberlain** [comment]
it does seem to work. i did 


```
pip install poetry
poetry run python app.py
```

**Comment by DouglasKrouth** [comment]
@josh-chamberlain Try running '''poetry install''' first

**Comment by josh-chamberlain** [comment]
@DouglasKrouth true, it worked but I think it was a fluke / because of my previous venv or something. added it to my previous comment

**Comment by mbodeantor** [comment]
Cool yeah! Sounds good to me

**Comment by DouglasKrouth** [comment]
@mbodeantor @josh-chamberlain so I'm thinking that a possible next step is to set up a short meeting for us to go over how this would affect the project structure, as well as any concerns you might have. If we decide to use _poetry_, I'll open a separate issue and link myself to update the docs to reflect how we want contributors to build the project going forward.

**Comment by josh-chamberlain** [comment]
@DouglasKrouth @mbodeantor I'm always happy to meet, but this seems fine to me—it's also easy enough to go back if there are issues. It's not making system level changes, just project-level environments, so it's safer than asking people to use anaconda or something. 

i'm going to update dev then make a PR for the poetry stuff into dev, just to test the config and see if it's working.

If you still want to meet, feel free to invite me and I'll try to make it! Marty also has my blessing to decide without me.

**Comment by mbodeantor** [comment]
@DouglasKrouth Same, happy to meet but likely whatever you add will be an improvement. My concerns are pretty base level, just started incorporating pytest but would like to incorporate it in the PR process so it blocks merge if tests fail

**Comment by mbodeantor** [comment]
@DouglasKrouth Are you familiar with deploying on Digital Ocean with Poetry? This is the most recent thing I've been able to find so far: https://www.digitalocean.com/community/questions/app-platform-python-poetry-support

**Comment by DouglasKrouth** [comment]
@mbodeantor that is actually a great catch and something that I hadn't considered before suggesting poetry. It appears as though BlueOcean doesn't have built in support for the tool. What's more surprising to me is that BlueOcean also lacks support for pyproject.toml format (which is surprising as it was introduced ~3 years ago [PEP 621](https://peps.python.org/pep-0621/)). 

Thinking an alternative to is just adding a few build/run scripts shell scripts if we're interested in a bit of the automation functionality that Poetry offers.

Solely because of the requirement to build/deploy on BlueOcean, I don't think Poetry would be a great choice for the data-sources-app at this time.

**Comment by mbodeantor** [comment]
@DouglasKrouth Okay no worries! Thanks for looking into it

**Comment by josh-chamberlain** [comment]
since we're using digitalocean's app platform, this isn't possible. if we ever switch to a self-managed hosting solution, we could do this. right now...not worth the devops. thanks for thinking of this and checking it out!

https://github.com/Police-Data-Accessibility-Project/data-sources-app/pull/112#issuecomment-1815173937
